### PR TITLE
whatfiles: init at 2020-06-16

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/whatfiles/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  buildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    install -Dm755 bin/whatfiles $out/bin/whatfiles
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Log what files are accessed by any Linux process";
+    homepage = "https://gitlab.com/spieglt/whatfiles";
```
Comment (doronbehar) [inline pkgs/tools/misc/whatfiles/default.nix:22]: 404

File: pkgs/tools/misc/whatfiles/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  version = "2020-06-16";
+
+  src = fetchFromGitHub {
+    owner = "spieglt";
+    repo = "whatfiles";
+    rev  = "2e12e73fec9683d95397b032fc3220ee00377ccc";
+    sha256 = "1l326ra8x38f3v99b7izcg6rwrl6dxps9xbp6ca0r711fdgaf1lg";
+  };
+
+  buildInputs = [ makeWrapper ];
```
Comment (doronbehar) [inline pkgs/tools/misc/whatfiles/default.nix:14]:
```suggestion
```

File: pkgs/tools/misc/whatfiles/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub, makeWrapper}:
+
+stdenv.mkDerivation {
+  pname = "whatfiles";
+  version = "2020-06-16";
```
Comment (mmahut) [inline pkgs/tools/misc/whatfiles/default.nix:5]: If this has no release, please use `unstable` flag.

Comment (matthiasbeyer) [reply pkgs/tools/misc/whatfiles/default.nix:5]: What `unstable` flag? You mean `version = "unstable-2020-06-16";`?

Comment (Ma27) [reply pkgs/tools/misc/whatfiles/default.nix:5]: Please set `pname` to `whatfiles-unstable` and simply use the date as `version`. See also #68518.

File: pkgs/tools/misc/whatfiles/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub, makeWrapper}:
```
Comment (tomberek) [inline pkgs/tools/misc/whatfiles/default.nix]:
```suggestion
{ lib, stdenv, fetchFromGitHub, makeWrapper}:
```

File: pkgs/tools/misc/whatfiles/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    repo = "whatfiles";
+    rev  = "2e12e73fec9683d95397b032fc3220ee00377ccc";
+    sha256 = "1l326ra8x38f3v99b7izcg6rwrl6dxps9xbp6ca0r711fdgaf1lg";
+  };
+
+  installPhase = ''
+    install -Dm755 bin/whatfiles $out/bin/whatfiles
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (tomberek) [inline pkgs/tools/misc/whatfiles/default.nix]:
```suggestion
  meta = with lib; {
```

**Review by tomberek — APPROVED** [review]

**Comment by tomberek** [comment]
Builds and runs as expected on NixOS. Applied small fixups and a rebase. Seems to have an error on aarch64.

File: pkgs/tools/misc/whatfiles/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, fetchFromGitHub, makeWrapper}:
+
+stdenv.mkDerivation {
+  pname = "whatfiles-unstable";
+  version = "2020-06-16";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/whatfiles/default.nix:5]:
```suggestion
  version = "unstable-2020-06-16";
```

File: pkgs/tools/misc/whatfiles/default.nix [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+  installPhase = ''
+    install -Dm755 bin/whatfiles $out/bin/whatfiles
+  '';
+
+  meta = with lib; {
+    description = "Log what files are accessed by any Linux process";
+    homepage = "https://github.com/spieglt/whatfiles";
+    maintainers = [ maintainers.matthiasbeyer ];
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/whatfiles/default.nix:22]: gpl3Only or gpl3Plus?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**